sources/ghostdog/src/main.rs

```diff
@@ -125,6 +168,51 @@ fn parse_device_name(device_info: &[u8], path: String) -> Result<String> {
         .to_string())
 }
 
+/// Read a file into a SupportedDevicesConfiguration Enum
+fn read_supported_devices_file(path: PathBuf) -> Result<SupportedDevicesConfiguration> {
+    let mut supported_devices_file =
+        fs::File::open(&path).context(error::OpenFileSnafu { path: path.clone() })?;
+    let mut supported_devices_str = String::new();
+    supported_devices_file
+        .read_to_string(&mut supported_devices_str)
+        .context(error::ReadFileSnafu { path: path.clone() })?;
+    let device_configuration: SupportedDevicesConfiguration =
+        serde_json::from_str(supported_devices_str.as_str())
+            .context(error::ParseGpuDevicesFileSnafu {})?;
+    Ok(device_configuration)
+}
+
+/// Search the Open GPU Supported Devices File to determine if the Open GPU Driver should be used based upon PCI devices present
+fn find_preferred_driver() -> Result<String> {
+    let open_gpu_devices = read_supported_devices_file(OPEN_GPU_SUPPORTED_DEVICES_PATH.into())?;
+    let list_input = pciclient::ListDevicesParam::builder()
+        .vendor(NVIDIA_VENDOR_ID)
+        .build();
+    let present_devices =
+        pciclient::list_devices(list_input).context(error::ListPciDevicesSnafu {})?;
```

Comment:
nit: for variant that only has `source` as member field, you can do 
```suggestion
        pciclient::list_devices(list_input).context(error::ListPciDevicesSnafu)?;
```

Same for L181